qiskit/transpiler/passes/layout/csp_layout.py

```diff
@@ -65,6 +66,32 @@ def recursiveBacktracking(self,  # pylint: disable=invalid-name
                                              single)
 
 
+class CustomConstraint(Constraint):
```

Comment:
Maybe rename this to be a bit more descriptive, something like `ValidEdgeConstrait` or something like that.